### PR TITLE
1832990: Add rhsm.no_insights config option, improve messaging

### PR DESCRIPTION
**Pull request by cnsnyder**
Improves messaging for auto insights register and unregister.
Now we check to see that insights-register.path is enabled before
messaging anything about automatic insights registration.

Likewise for unregistration.

### CODE REVIEW

File: src/subscription_manager/managercli.py [hunk 1]
```diff
@@ -1657,15 +1660,16 @@ def _do_command(self):
 
         if self.options.no_insights:
```
Comment (jirihnidek) [inline src/subscription_manager/managercli.py:1661]: Here should be: `if no_insights is True:`

File: test/test_utils.py [hunk 2]
```diff
@@ -989,3 +991,118 @@ def test_get_process_names(self):
...
+    """
+
+    def setUp(self):
+        check_returncode_patch = patch('subscription_manager.utils.check_returncode')
+        self.check_returncode_mock = check_returncode_patch.start()
+        self.addCleanup(check_returncode_patch.stop)
+
+    def test_is_insights_register_enabled(self):
+        self.check_returncode_mock.return_value = 0
+        self.assertTrue(is_insights_installed())
```
Comment (jirihnidek) [inline test/test_utils.py:1033]: When the name of this unit test is `test_is_insights_register_enabled`, then I expect the method `is_insights_register_enabled` will be called. This was probably caused by some copy-paste. I would change the name of the method to something like: `test_is_insights_client_installed`

File: test/test_utils.py [hunk 3]
```diff
@@ -989,3 +991,118 @@ def test_get_process_names(self):
...
+        self._assert_called_correct_cmd()
+
+    def test_is_insights_register_disabled(self):
```
Comment (jirihnidek) [inline test/test_utils.py:1036]: Similar issue here. I would change the name of the method to something like: `test_is_insights_client_not_installed`

**Comment by cnsnyder** [comment]
Merging based on jirihnidek's ACK

**Pull request merged**